EAD-Session-9-Assignment/src/main/java/com/metacube/assignment/controller/AppController.java

```diff
@@ -0,0 +1,154 @@
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import com.metacube.assignment.dto.EmployeeRegistrationForm;
+import com.metacube.assignment.dto.EmployeeLoginForm;
+import com.metacube.assignment.dto.GetPass;
+import com.metacube.assignment.dto.VehicleRegistrationForm;
+import com.metacube.assignment.model.Employee;
+
+/**
+ * This class represents a controller
+ * @author Jugal
+ * Dated 09/11/2019
+ */
+@Controller
+public class AppController {
+
+	// show employee registration form
+	@GetMapping("/employeeRegistration")
+	public String showEmployeeRegistrationForm(Model model) {
+		model.addAttribute("employeeRegistrationForm",
+				new EmployeeRegistrationForm());
+		return "users/employee_registration_form";
+	}
+
+	// method to register employee
+	@PostMapping("/employeeRegistration")
+	public String doEmployeeRegister(
+			@Validated EmployeeRegistrationForm registeringEmployee,
+			BindingResult result) {
+		if (result.hasErrors()) {
+			ObjectError objectError = new ObjectError("form",
+					"Please fill all the details");
+			result.addError(objectError);
+			return "users/employee_registration_form";
+		} else {
+			return "redirect:/vehicleRegistration";
+		}
+	}
+
+	// show employee login form
+	@GetMapping("/employeeLogin")
+	public String showEmployeeLoginForm(Model model) {
+		model.addAttribute("employeeLoginForm", new EmployeeLoginForm());
+		return "users/employee_login_form";
+	}
+
+	// Method to login employee
+	@PostMapping("/employeeLogin")
+	public String doSignIn(@Validated EmployeeLoginForm loggingEmployee,
+			BindingResult result) {
+		if (result.hasErrors()) {
+			ObjectError objectError = new ObjectError("form",
+					"Please fill all the details");
+			result.addError(objectError);
+			return "users/employee_login_form";
+		} else {
+			return "redirect:/home";
+		}
+	}
+
+	// show vehicle registration form
+	@GetMapping("/vehicleRegistration")
+	public String showVehicleRegistrationForm(Model model) {
+		model.addAttribute("vehicleRegistrationForm",
+				new VehicleRegistrationForm());
+		return "users/vehicle_registration_form";
+	}
+
+	// Method to register vehicle
+	@PostMapping("/vehicleRegistration")
+	public String doVehicleRegister(
+			@Validated VehicleRegistrationForm registeringVehicle,
+			BindingResult result, Model model) {
+		if (result.hasErrors()) {
+			ObjectError objectError = new ObjectError("form",
+					"Please fill all the details");
+			result.addError(objectError);
+			return "users/vehicle_registration_form";
+		} else {
+			String selectedVehicleType = registeringVehicle.getType();
+			model.addAttribute("selectedVehicleType", selectedVehicleType);
+			model.addAttribute("getPass", new GetPass());
+			return "users/pass_details";
+		}
+	}
+
+	// show friends list
+	@GetMapping("/home/friendsList")
+	public String showFriendsList(Model model) {
+		List<Employee> employeeList = new ArrayList<>();
```

Comment:
Move the list out of the method. In this case, it will create a new list for every method call.

Reply:
Done